store/rootmulti/store.go

```diff
@@ -458,6 +458,13 @@
 	return rs.lastCommitInfo.CommitID()
 }
 
+// SetCommitting implements Committer/CommitStore.
+func (rs *Store) SetCommitting(committing bool) {
+	for _, store := range rs.stores {
+		store.SetCommitting(committing)
+	}
```

Comment:
Did you consider passing in a shared lock to the IAVL constructor in order to avoid this iteration step and the interface bloat & abstraction leaking of `SetComitting`?